common/stage_timer.go

```diff
@@ -0,0 +1,100 @@
+package common
+
+import (
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+)
+
+// It may be helpful to generalize this utility to be used in other parts of the codebase. For a future PR, perhaps.
+
+// StageTimer encapsulates metrics to help track the time spent in each stage of the payload dispersal process.
+type StageTimer struct {
+	stageCount   *prometheus.GaugeVec
+	stageLatency *prometheus.SummaryVec
+}
+
+// SequenceProbe tracks the timing of a single sequence of operations. Multiple sequences can be tracked concurrently.
+type SequenceProbe struct {
+	stageTimer        *StageTimer
+	currentStage      string
+	currentStageStart time.Time
+}
+
+// NewStageTimer creates a new stageTimer with the given prefix and name.
+func NewStageTimer(registry *prometheus.Registry, prefix, name string) *StageTimer {
+	if registry == nil {
+		return nil
+	}
+
+	statusLatency := promauto.With(registry).NewSummaryVec(
+		prometheus.SummaryOpts{
+			Namespace:  prefix,
+			Name:       name + "_stage_latency_ms",
+			Help:       "the latency of each type of operation",
+			Objectives: map[float64]float64{0.5: 0.05, 0.9: 0.01, 0.99: 0.001},
+		},
+		[]string{"stage"},
+	)
+
+	statusCount := promauto.With(registry).NewGaugeVec(
```

Comment:
mixed terminology, sometimes "statusCount", sometimes "stageCount". Is there any reason that both "stage" and "status" need to exist in this file, it seems both refer to the same concept

Reply:
`status` -> `stage`